src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionTransfer.java

```diff
@@ -41,8 +48,47 @@ public TransferResult<CFValue, CFStore> visitMethodInvocation(
               boundClassTypeMirror.getAnnotation(), diATF.classValValueElement, String.class);
 
       accumulate(node, result, classNames.toArray(new String[1]));
+    } else if (diATF.isAnnotatedWithMethod(node.getTree())) {
+      MethodInvocationNode methodInvocationNode = node;
+      MethodInvocationNode callToBindNode =
+          (MethodInvocationNode) methodInvocationNode.getTarget().getReceiver();
+      MethodInvocationNode namesClassNode =
+          (MethodInvocationNode) methodInvocationNode.getArgument(0);
+      FieldAccessNode boundClass = (FieldAccessNode) callToBindNode.getArgument(0);
+      StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
+
+      accumulateBindAnnotatedWith(
+          node, result, boundClass.getReceiver().getType().toString(), givenName.getValue());
     }
 
     return result;
   }
+
+  public void accumulateBindAnnotatedWith(
+      Node node, TransferResult<CFValue, CFStore> result, String value, String name) {
+    List<String> valuesAsList = Arrays.asList(value);
+    List<String> namesAsList = Arrays.asList(name);
+    // If dataflow has already recorded information about the target, fetch it and integrate
+    // it into the list of values in the new annotation.
+    JavaExpression target = JavaExpression.fromNode(node);
+
+    AnnotationMirror newAnno = diATF.createBindAnnotatedWithAnnotation(valuesAsList, namesAsList);
+
+    // TODO: insertValue not updating stores, insertValuePermitNondeterministic is
+    if (result.containsTwoStores()) {
+      result.getThenStore().insertValuePermitNondeterministic(target, newAnno);
+      result.getElseStore().insertValuePermitNondeterministic(target, newAnno);
+    } else {
+      result.getRegularStore().insertValuePermitNondeterministic(target, newAnno);
+    }
+
+    Tree tree = node.getTree();
+
+    if (tree != null && tree.getKind() == Tree.Kind.METHOD_INVOCATION) {
+      Node receiver = ((MethodInvocationNode) node).getTarget().getReceiver();
+      if (receiver != null && diATF.returnsThis((MethodInvocationTree) tree)) {
+        accumulateBindAnnotatedWith(receiver, result, value, name);
+      }
+    }
```

Comment:
This code propagates the annotation back through a fluent chain, if each element in the fluent chain returns its receiver instead of a new object. I don't think we actually want this behavior (or at least, I don't think we need it, so keeping it seems unnecessary), but I'm not sure.

Reply:
This decision still needs to be taken.

Reply:
I'm not sure what the best course of action for this is. I think you should make the call on this one.

Reply:
I would remove it.